internal/provider/resource_apply_module.go

```diff
@@ -433,54 +515,81 @@ func (t *applyModuleResource) doRun(ctx context.Context,
 	})
 	if err != nil {
 		diags.AddError("Failed to apply a run", err.Error())
-		return diags
+		return nil, diags
 	}
 
 	// Make sure the run has an apply.
 	if appliedRun.Apply == nil {
 		msg := fmt.Sprintf("Created run does not have an apply: %s", appliedRun.Metadata.ID)
 		diags.AddError(msg, "")
-		return diags
+		return nil, diags
 	}
 
 	if err = t.waitForJobCompletion(ctx, appliedRun.Apply.CurrentJobID); err != nil {
 		diags.AddError("Failed to wait for apply job completion", err.Error())
-		return diags
+		return nil, diags
 	}
 
 	finishedRun, err := t.client.Run.GetRun(ctx, &sdktypes.GetRunInput{ID: appliedRun.Metadata.ID})
 	if err != nil {
 		diags.AddError("Failed to get finished run", err.Error())
-		return diags
+		return nil, diags
 	}
 
 	// If an apply job succeeds, finishedRun.Status is "applied" and
 	// finishedRun.Apply.Status is "finished".
 	if finishedRun.Status != "applied" {
+
+		// Bring in any error message(s) from the finished inner apply run.
+		innerApplyRunDiags := t.extractRunError(ctx, finishedRun)
+		diags.Append(innerApplyRunDiags...)
+		if innerApplyRunDiags.HasError() {
+			return nil, diags
+		}
+
 		diags.AddError("Apply failed", string(finishedRun.Status))
-		return diags
+		return nil, diags
 	}
 	if finishedRun.Apply.Status != "finished" {
+
+		// Bring in any error message(s) from the finished inner apply run.
+		innerApplyRunDiags := t.extractRunError(ctx, finishedRun)
```

Comment:
This code block appears to be the duplicating the same logic from the IF condition above.